gazebo-classic "C++ CI Workflow with conda-forge dependencies" job failing since two weeks ago on macOS

Last successful job: https://github.com/gazebosim/gazebo-classic/actions/runs/4549669663
First failing job: https://github.com/gazebosim/gazebo-classic/actions/runs/4559987576
Last (at the moment of the opening of the issue) failing job: https://github.com/gazebosim/gazebo-classic/actions/runs/4705540516

The failure is: 
~~~
In file included from /Users/runner/miniconda3/envs/test/include/boost/type_index.hpp:29:
In file included from /Users/runner/miniconda3/envs/test/include/boost/type_index/stl_type_index.hpp:47:
/Users/runner/miniconda3/envs/test/include/boost/container_hash/hash.hpp:131:33: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
        struct hash_base : std::unary_function<T, std::size_t> {};
                           ~~~~~^
/Users/runner/miniconda3/envs/test/bin/../include/c++/v1/__functional/unary_function.h:46:1: note: '__unary_function' declared here
using __unary_function = __unary_function_keep_layout_base<_Arg, _Result>;
^
1 error generated.
~~~

So, the issue is related to https://github.com/conda-forge/boost-cpp-feedstock/issues/134, that was fixed for boost 1.78 . However, for some reason that I do not fully understand we are using an older boost in the CI.

--- a/.github/workflows/conda-forge.yml
+++ b/.github/workflows/conda-forge.yml
@@ -36,7 +36,7 @@ jobs:
         # Compilation related dependencies
         mamba install cmake compilers make ninja pkg-config
         # Actual dependencies
-        mamba install libprotobuf libsdformat libignition-cmake2 libignition-math6 libignition-transport8 libignition-common3 libignition-fuel-tools4 qt-main ogre=1.10 freeimage curl tbb-devel qwt tinyxml2 libccd boost-cpp libcurl tinyxml bzip2 zlib ffmpeg graphviz libgdal libusb bullet-cpp simbody hdf5 openal-soft glib gts dartsim
+        mamba install libprotobuf libsdformat libignition-cmake2 libignition-math6 libignition-transport8 libignition-common3 libignition-fuel-tools4 qt-main ogre=1.10 freeimage curl tbb-devel qwt tinyxml2 libccd-double boost-cpp libcurl tinyxml bzip2 zlib ffmpeg graphviz libgdal libusb bullet-cpp simbody hdf5 openal-soft glib gts dartsim
 
     - name: Linux-only Dependencies [Linux]
       if: contains(matrix.os, 'ubuntu')

